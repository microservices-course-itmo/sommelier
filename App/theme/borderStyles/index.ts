export const borderStyles = {
  solid: 'solid',
  dashed: 'dashed',
  dotted: 'dotted',
}
