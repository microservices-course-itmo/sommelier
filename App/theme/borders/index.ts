export const borders = {
  dotted: 'dotted',
  dashed: 'dashed',
  solid: 'solid',
}
