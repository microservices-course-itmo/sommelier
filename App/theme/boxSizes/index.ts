export const SIZE = {
  xs: 'xs',
  s: 's',
  m: 'm',
  l: 'l',
  xl: 'xl',
} as const

export const boxSizes = {
  xs: '4px',
  s: '8px',
  m: '16px',
  l: '24px',
  xl: '32px',
} as const
