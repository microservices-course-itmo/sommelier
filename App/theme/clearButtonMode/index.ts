export const clearButtonMode = {
  never: 'never',
  always: 'always',
} as const

export const CLEAR_BUTTON_MODE_KEYS = {
  never: 'never',
  always: 'always',
} as const
