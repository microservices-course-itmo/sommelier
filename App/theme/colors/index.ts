export const colors = {
  transparent: '#00000000',
  white: '#fff',
  black: '#000',
  lightGray: '#fafbfc',
  lightGrayBorder: '#e1e4e8',
  lightGrayText: '#586069',
  grayBackground: '#f6f8fa',
  grayBorder: '#b9b9b9',
  primary: '#0265d6',
  dark: '#0366d6',
  light: '#58a5f0',
  fillRed: '#A22841',
  fillGray: '#757575',
  primaryRed: '#931332',
  primaryGray: '#666666',
  primaryDarkRed: '#680e24',
  shadowButton: 'rgba(104, 14, 36, 0.25)',
  notEditableField: 'rgba(127, 127, 127, 0.3)',
  searchInputBorder: '#9e9e9e',
  priceInputBorder: '#dcdcdc',
  priceInputPlaceholedText: '#B5B5B5',
  priceInputText: '#333333',

  switchToggleOn: 'rgba(69, 208, 158, 0.5)',
  switchToggleOff: '#ededed',
  switchNotchOn: '#45d09e',
  switchNotchOff: '#e0e0e0',
} as const

export const COLOR_KEYS = {
  transparent: 'transparent',
  white: 'white',
  black: 'black',
  lightGray: 'lightGray',
  lightGrayBorder: 'lightGrayBorder',
  lightGrayText: 'lightGrayText',
  grayBackground: 'grayBackground',
  grayBorder: 'grayBorder',
  primary: 'primary',
  dark: 'dark',
  light: 'light',
  fillRed: 'fillRed',
  fillGray: 'fillGray',
  primaryRed: 'primaryRed',
  primaryGray: 'primaryGray',
  primaryDarkRed: 'primaryDarkRed',
  shadowButton: 'shadowButton',
  notEditableField: 'notEditableField',
  searchInputBorder: 'searchInputBorder',
  priceInputBorder: 'priceInputBorder',
  priceInputPlaceholedText: 'priceInputPlaceholedText',
  priceInputText: 'priceInputText',

  switchToggleOn: 'switchToggleOn',
  switchToggleOff: 'switchToggleOff',
  switchNotchOn: 'switchNotchOn',
  switchNotchOff: 'switchNotchOff',
} as const
