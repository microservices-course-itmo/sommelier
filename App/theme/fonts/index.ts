export const fonts = {
  PtSans: 'Playfair Display',
  PlayfairDisplay: 'Playfair Display',
  Montserrat: 'Montserrat',
}
