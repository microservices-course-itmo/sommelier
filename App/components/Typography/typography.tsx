import React, { ReactNode } from 'react'
import * as Styled from './typography.styled'
import theme from '../../theme/index'
import { ReactElement } from 'react'

type Props = {
  children: ReactNode
  color: string
  bold?: boolean
  italic: boolean
  size?: keyof typeof theme.fontSizes
  crossedOut?: boolean
  fontFamily: string
}

export const H1 = ({ color, children, bold, italic, fontFamily }: Props) => (
  <Styled.StyledH1
    color={color}
    bold={bold}
    italic={italic}
    fontFamily={fontFamily}
  >
    {children}
  </Styled.StyledH1>
)

export const H2 = ({ color, children, bold, italic, fontFamily }: Props) => (
  <Styled.StyledH2
    color={color}
    bold={bold}
    italic={italic}
    fontFamily={fontFamily}
  >
    {children}
  </Styled.StyledH2>
)

export const H3 = ({ color, children, bold, italic, fontFamily }: Props) => (
  <Styled.StyledH3
    color={color}
    bold={bold}
    italic={italic}
    fontFamily={fontFamily}
  >
    {children}
  </Styled.StyledH3>
)

export const H4 = ({ color, children, bold, italic, fontFamily }: Props) => (
  <Styled.StyledH4
    color={color}
    bold={bold}
    italic={italic}
    fontFamily={fontFamily}
  >
    {children}
  </Styled.StyledH4>
)

export const H5 = ({ color, children, bold, italic, fontFamily }: Props) => (
  <Styled.StyledH5
    color={color}
    bold={bold}
    italic={italic}
    fontFamily={fontFamily}
  >
    {children}
  </Styled.StyledH5>
)

export const H6 = ({ color, children, bold, italic, fontFamily }: Props) => (
  <Styled.StyledH6
    color={color}
    bold={bold}
    italic={italic}
    fontFamily={fontFamily}
  >
    {children}
  </Styled.StyledH6>
)

export const Text = ({
  children,
  color,
  bold,
  italic,
  size,
  crossedOut,
  fontFamily,
}: Props): ReactElement => (
  <Styled.StyledText
    color={color}
    bold={bold}
    italic={italic}
    fontFamily={fontFamily}
    size={size}
    crossedOut={crossedOut}
  >
    {children}
  </Styled.StyledText>
)
