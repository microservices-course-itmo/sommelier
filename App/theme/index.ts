import { headerSizes } from './headerSizes'
import { boxSizes } from './boxSizes'
import { colors } from './colors'
import { fontSizes } from './fontSizes'
import { borders } from './borders'
import { inputSizes, searchSizes, inputFontSizes } from './inputSearchSizes'

const theme = {
  boxSizes: {
    ...boxSizes,
  },
  headerSizes: {
    ...headerSizes,
  },
  colors: {
    ...colors,
  },
  fontSizes: {
    ...fontSizes,
  },
  borders: {
    ...borders,
  },
  inputSizes: {
    ...inputSizes,
  },
  searchSizes: {
    ...searchSizes,
  },
  inputFontSizes: {
    ...inputFontSizes,
  },
}

export default theme
