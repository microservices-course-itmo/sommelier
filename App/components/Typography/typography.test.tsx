import { render } from '@testing-library/react-native'
import React from 'react'

import { H1, H2, H3, H4, H5, H6, Text } from './typography'
import theme from '../../theme'

const defaultProps = {
  fontFamily: theme.fonts.PtSans
}

describe("Typography component", () => {
  describe("Component matches snapshot", () => {
    it("Text renders correctly", () => {
      const tree = render(<Text {...defaultProps}>abc 123</Text>).toJSON();
      expect(tree).toMatchSnapshot();
    })
    it("H1 renders correctly", () => {
      const tree = render(<H1 {...defaultProps}>abc 123</H1>).toJSON();
      expect(tree).toMatchSnapshot();
    })
    it("H2 renders correctly", () => {
      const tree = render(<H2 {...defaultProps}>abc 123</H2>).toJSON();
      expect(tree).toMatchSnapshot();
    })
    it("H3 renders correctly", () => {
      const tree = render(<H3 {...defaultProps}>abc 123</H3>).toJSON();
      expect(tree).toMatchSnapshot();
    })
    it("H4 renders correctly", () => {
      const tree = render(<H4 {...defaultProps}>abc 123</H4>).toJSON();
      expect(tree).toMatchSnapshot();
    })
    it("H5 renders correctly", () => {
      const tree = render(<H5 {...defaultProps}>abc 123</H5>).toJSON();
      expect(tree).toMatchSnapshot();
    })
    it("H6 renders correctly", () => {
      const tree = render(<H6 {...defaultProps}>abc 123</H6>).toJSON();
      expect(tree).toMatchSnapshot();
    })
  })
})