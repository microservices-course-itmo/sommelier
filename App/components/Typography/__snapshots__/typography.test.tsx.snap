// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text component Component matches snapshot H1 renders correctly 1`] = `
<Text
  color="black"
  fontFamily="PtSans"
  italic={false}
  style={
    Array [
      Object {
        "color": "black",
        "fontFamily": "PtSans",
        "fontSize": 32,
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  abc 123
</Text>
`;

exports[`Text component Component matches snapshot H2 renders correctly 1`] = `
<Text
  color="black"
  fontFamily="PtSans"
  italic={false}
  style={
    Array [
      Object {
        "color": "black",
        "fontFamily": "PtSans",
        "fontSize": 24,
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  abc 123
</Text>
`;

exports[`Text component Component matches snapshot H3 renders correctly 1`] = `
<Text
  color="black"
  fontFamily="PtSans"
  italic={false}
  style={
    Array [
      Object {
        "color": "black",
        "fontFamily": "PtSans",
        "fontSize": 18.72,
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  abc 123
</Text>
`;

exports[`Text component Component matches snapshot H4 renders correctly 1`] = `
<Text
  color="black"
  fontFamily="PtSans"
  italic={false}
  style={
    Array [
      Object {
        "color": "black",
        "fontFamily": "PtSans",
        "fontSize": 16,
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  abc 123
</Text>
`;

exports[`Text component Component matches snapshot H5 renders correctly 1`] = `
<Text
  color="black"
  fontFamily="PtSans"
  italic={false}
  style={
    Array [
      Object {
        "color": "black",
        "fontFamily": "PtSans",
        "fontSize": 13.28,
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  abc 123
</Text>
`;

exports[`Text component Component matches snapshot H6 renders correctly 1`] = `
<Text
  color="black"
  fontFamily="PtSans"
  italic={false}
  style={
    Array [
      Object {
        "color": "black",
        "fontFamily": "PtSans",
        "fontSize": 10.72,
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  abc 123
</Text>
`;

exports[`Text component Component matches snapshot Renders correctly 1`] = `
<Text
  color="white"
  fontFamily="PtSans"
  italic={false}
  style={
    Array [
      Object {
        "color": "white",
        "fontFamily": "PtSans",
        "fontSize": 12,
        "fontStyle": "normal",
        "fontWeight": "normal",
        "textDecorationColor": "black",
        "textDecorationLine": "none",
        "textDecorationStyle": "solid",
      },
    ]
  }
>
  abc
</Text>
`;

exports[`Text component Component matches snapshot Text renders correctly 1`] = `
<Text
  color="black"
  fontFamily="PtSans"
  italic={false}
  style={
    Array [
      Object {
        "color": "black",
        "fontFamily": "PtSans",
        "fontSize": 12,
        "fontStyle": "normal",
        "fontWeight": "normal",
        "textDecorationColor": "black",
        "textDecorationLine": "none",
        "textDecorationStyle": "solid",
      },
    ]
  }
>
  abc 123
</Text>
`;

exports[`Typography component Component matches snapshot H1 renders correctly 1`] = `
<Text
  color="black"
  fontFamily="PtSans"
  italic={false}
  style={
    Array [
      Object {
        "color": "black",
        "fontFamily": "PtSans",
        "fontSize": 32,
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  abc 123
</Text>
`;

exports[`Typography component Component matches snapshot H2 renders correctly 1`] = `
<Text
  color="black"
  fontFamily="PtSans"
  italic={false}
  style={
    Array [
      Object {
        "color": "black",
        "fontFamily": "PtSans",
        "fontSize": 24,
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  abc 123
</Text>
`;

exports[`Typography component Component matches snapshot H3 renders correctly 1`] = `
<Text
  color="black"
  fontFamily="PtSans"
  italic={false}
  style={
    Array [
      Object {
        "color": "black",
        "fontFamily": "PtSans",
        "fontSize": 18.72,
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  abc 123
</Text>
`;

exports[`Typography component Component matches snapshot H4 renders correctly 1`] = `
<Text
  color="black"
  fontFamily="PtSans"
  italic={false}
  style={
    Array [
      Object {
        "color": "black",
        "fontFamily": "PtSans",
        "fontSize": 16,
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  abc 123
</Text>
`;

exports[`Typography component Component matches snapshot H5 renders correctly 1`] = `
<Text
  color="black"
  fontFamily="PtSans"
  italic={false}
  style={
    Array [
      Object {
        "color": "black",
        "fontFamily": "PtSans",
        "fontSize": 13.28,
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  abc 123
</Text>
`;

exports[`Typography component Component matches snapshot H6 renders correctly 1`] = `
<Text
  color="black"
  fontFamily="PtSans"
  italic={false}
  style={
    Array [
      Object {
        "color": "black",
        "fontFamily": "PtSans",
        "fontSize": 10.72,
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  abc 123
</Text>
`;

exports[`Typography component Component matches snapshot Text renders correctly 1`] = `
<Text
  color="black"
  fontFamily="PtSans"
  italic={false}
  style={
    Array [
      Object {
        "color": "black",
        "fontFamily": "PtSans",
        "fontSize": 12,
        "fontStyle": "normal",
        "fontWeight": "normal",
        "textDecorationColor": "black",
        "textDecorationLine": "none",
        "textDecorationStyle": "solid",
      },
    ]
  }
>
  abc 123
</Text>
`;
